pkg/controller/tektoninstallation/tektoninstallation_controller.go

```diff
@@ -87,41 +106,129 @@ func (r *ReconcileTektonInstallation) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
-	err := r.EnsureTektonSubscription(reqLogger, tektonInstallation)
-	return reconcile.Result{}, err
+	if created, err := r.EnsureTektonSubscription(reqLogger, tektonInstallation); err != nil {
+		return reconcile.Result{}, err
+	} else if created {
+		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonSubscriptionCreated, "created tekton subscription")
+	}
+
+	if requeue, err := r.ensureWatchTektonConfig(); err != nil {
+		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationFailed, err, "failed to add watch for TektonConfig")
+	} else if requeue {
+		return reconcile.Result{Requeue: true, RequeueAfter: 3 * time.Second}, nil
+	}
+
+	tektonCfg := &config.Config{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: TektonConfigName}, tektonCfg)
+	if err != nil {
+		reqLogger.Info("requeue to retrieve config resource", "tektonconfigname", TektonConfigName)
+		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationFailed, err, "failed to add get TektonConfig")
+	}
+
+	switch getTektonConfigStatus(tektonCfg) {
+	case config.InstalledStatus:
+		reqLogger.Info("done with Tekton installation")
+		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationSucceeded, "tekton installation succeeded")
+	case config.InstallingStatus:
+		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationInstalling, "tekton installation installing")
+	case config.ErrorStatus:
+		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationFailed, "tekton installation failed with error")
```

Comment:
it would be great if you could take the `ConfigCondition.Details` and add it as a message
https://github.com/tektoncd/operator/blob/master/pkg/apis/operator/v1alpha1/config_types.go#L33
So we know what is going on